test/blackbox/PubSubWriterReader.hpp

```diff
@@ -64,28 +71,151 @@ class PubSubWriterReader
                 }
             }
 #endif
+            void onParticipantDiscovery(
+                    eprosima::fastrtps::Participant* participant,
+                    eprosima::fastrtps::rtps::ParticipantDiscoveryInfo&& info) override
+            {
+                (void)participant;
+
+                switch (info.status)
+                {
+                    case eprosima::fastrtps::rtps::ParticipantDiscoveryInfo::DISCOVERED_PARTICIPANT:
+                        info_add(discovered_participants_, info.info.m_guid);
+                        break;
+
+                    case eprosima::fastrtps::rtps::ParticipantDiscoveryInfo::REMOVED_PARTICIPANT:
+                        info_remove(discovered_participants_, info.info.m_guid);
+                        break;
+
+                    case eprosima::fastrtps::rtps::ParticipantDiscoveryInfo::DROPPED_PARTICIPANT:
+                        // TODO: Should we warn here?
```

Comment:
Remove TODO and maybe add info prints about the participant discovered.

Reply:
done!